app/lib/hooks/useOffsetParams.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import {
+  InputToken,
+  inputTokens,
+  RetirementToken,
+  retirementTokens,
+} from "@klimadao/lib/constants";
+
+const inputParams = [
+  "quantity",
+  "inputToken",
+  "retirementToken",
+  "beneficiary",
+  "beneficiaryAddress",
+  "message",
+  "projectToken",
```

Comment:
true, I was just overthinking it because each query entry expects a single value
`projectToken=0x123&projectToken=0x123`
but the output downstream from `params.getAll` is an array